toolchain/parse/typed_nodes.h

```diff
@@ -127,10 +127,17 @@ using FileEnd = LeafNode<NodeKind::FileEnd, Lex::FileEndTokenIndex>;
 using EmptyDecl = LeafNode<NodeKind::EmptyDecl, Lex::SemiTokenIndex,
                            NodeCategory::Decl | NodeCategory::Statement>;
 
-// A name in a non-expression context, such as a declaration.
-using IdentifierName =
-    LeafNode<NodeKind::IdentifierName, Lex::IdentifierTokenIndex,
-             NodeCategory::MemberName>;
+// A name in a non-expression context, such as a declaration, that is known
+// to be followed by parameters.
+using IdentifierNameBeforeParams =
+    LeafNode<NodeKind::IdentifierNameBeforeParams, Lex::IdentifierTokenIndex,
+             NodeCategory::MemberName | NodeCategory::NonExprIdentifierName>;
```

Comment:
How come `IdentifierName` was not `NodeCategory::NonExprIdentifierName` before, but this is now?

Reply:
FWIW, one thing to mull might be if `IdentifierNameExpr` could actually be `NameRef` (since it looks like it's always translated to `SemIR::NameRef`), which might avoid the ambiguity of this

Reply:
True, although `SemIR::NameRef`s can also be created from other node kinds, so they don't line up perfectly. Definitely an option to consider, though.